### PR TITLE
Refactored Rename Tab dialog and its context menu

### LINKED ISSUE #2709: Show context in rename dialog
**What should be added?**
When renaming a tab, a simple dialog appears with a single text box. This should explain to the user that they are renaming the tab as well as showing the name of the tab in the title/text box placeholder.

**Why should it be added?**
It is not obvious what the dialog does if a user opens it by accident. You could also misclick and believe that you're renaming a different tab than the one being acted on.

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

I removed `TextInputDialog` class since it was only useful in `NotebookTab.cpp` and adding to that class was more expensive than constructing `QDialog` object inside `NotebookTab.cpp`.

- Added placeholder to lineEdit which shows default tab name.
- Added some spacing, a label and renamed window title to make things look cleaner.
- Edited Tab context menu to make it easier to understand.

layouts are dank my transparency is even danker

Closes #2709

### CODE REVIEW

**Comment by jammehcow** [comment]
Tested, LGTM 👍

**Review by jammehcow — APPROVED** [review]

**Comment by fourtf** [comment]
Even though this is the suggested solution from issue thread I'm not sure if this solves the issue accurately. A placeholder should help the user understand what he has to input and I don't think that showing up he channel name does that.

**Comment by fourtf** [comment]
I played around a bit in figma and came up with this:
image

It incorporates some other elements to make it more clear what the dialog is for and what the user has to input. I looked at the right click menu of the tab and tried to improve on it a bit to also make it more clear there.

#2709

**Comment by zneix** [comment]
What I managed to achieve:

layouts are dank my transparency is even danker

File: src/widgets/helper/NotebookTab.cpp [hunk 1]
```diff
@@ -75,15 +74,53 @@ NotebookTab::NotebookTab(Notebook *notebook)
...
+    lineEdit->setText(this->getCustomTitle());
+    lineEdit->setPlaceholderText(this->getDefaultTitle());
+    lineEdit->selectAll();
+
+    vbox->addWidget(new QLabel("Name:"));
+    vbox->addWidget(lineEdit);
+    vbox->addStretch(1);
+
+    auto okButton = new QPushButton("OK");
+    auto cancelButton = new QPushButton("Canel");
```
Comment (Mm2PL) [inline src/widgets/helper/NotebookTab.cpp:91]: canel forsenT

**Review by Mm2PL — CHANGES_REQUESTED** [review]

File: src/widgets/helper/NotebookTab.cpp [hunk 2]
```diff
@@ -75,15 +74,53 @@ NotebookTab::NotebookTab(Notebook *notebook)
...
+    lineEdit->selectAll();
+
+    vbox->addWidget(new QLabel("Name:"));
+    vbox->addWidget(lineEdit);
+    vbox->addStretch(1);
+
+    auto okButton = new QPushButton("OK");
+    auto cancelButton = new QPushButton("Cancel");
+
+    auto buttonBox = new QHBoxLayout;
```
Comment (pajlada) [inline src/widgets/helper/NotebookTab.cpp:93]: Use a https://doc.qt.io/qt-5/qdialogbuttonbox.html instead

**Review by pajlada — APPROVED** [review]
LGTM :+1: Merge in after adding the changelog entry as agreed

**Pull request merged**